### PR TITLE
Show Extreme Notes Per Voice

### LINKED ISSUE #2: Highlight highest and lowest notes

### PR DESCRIPTION
**Pull request by curran**
Will close #2 . Same changes as in #38, but rebased off #41 with merge conflicts resolved.

### CODE REVIEW

**Comment by curran** [comment]
@seemantk IMO this is demo-able and ready to merge as-is. I have created new issues for follow-on work that can go in separate PRs:
- Add UI to toggle extreme note highlights #45
- Highlight globally highest and lowest notes #44 
- Fix Y Scale Domain in Focus View #43

**Comment by curran** [comment]
What it looks like:

image

image

File: js/notescanvas.js [hunk 1]
```diff
@@ -13,6 +13,11 @@ function NotesCanvas() {
       , roundedCornerSize
       , dispatch
       , state = true // on; false = off
+      , showExtremeNotes = false
+      , isExtremeNote = d3.local()
```
Comment (seemantk) [inline js/notescanvas.js:17]: Neat!

File: js/notescanvas.js [hunk 2]
```diff
@@ -91,6 +112,17 @@ function NotesCanvas() {
...
     } // setHeights()
 
+    function computeExtremeNotes() {
+        var notes = svg.datum().value
+          , pitchExtent = d3.extent(notes, function (d){ return d.pitch; })
+        ;
+        svg.selectAll("rect.note")
+            .each(function (d){
+                isExtremeNote.set(this, d.pitch === pitchExtent[0] || d.pitch === pitchExtent[1]);
+              })
```
Comment (seemantk) [inline js/notescanvas.js:122]: Nifty to set a datum this way!  Nice one, @curran

**Pull request merged**